var/spack/repos/builtin/packages/py-geopandas/package.py

```diff
@@ -48,15 +50,21 @@ class PyGeopandas(PythonPackage):
     depends_on("py-pandas@0.25.0:", type=("build", "run"), when="@0.10:")
     depends_on("py-pandas@1.0.0:", type=("build", "run"), when="@0.11:")
     depends_on("py-pandas@1.4.0:", type=("build", "run"), when="@0.14:")
-    depends_on("py-shapely@:1", type=("build", "run"))
+    depends_on("py-shapely@:1", type=("build", "run"), when="@:0.99")
     depends_on("py-shapely@1.6:1", type=("build", "run"), when="@0.9:0.10")
-    depends_on("py-shapely@1.7:1", type=("build", "run"), when="@0.11:")
+    depends_on("py-shapely@1.7:1", type=("build", "run"), when="@0.11:0.99")
     depends_on("py-shapely@1.8.0:", type=("build", "run"), when="@0.14:")
-    depends_on("py-fiona", type=("build", "run"))
-    depends_on("py-fiona@1.8:", type=("build", "run"), when="@0.9:")
-    depends_on("py-fiona@1.8.21:", type=("build", "run"), when="@0.14:")
+    depends_on("py-shapely@2.0.0:", type=("build", "run"), when="@1:")
     depends_on("py-pyproj", type=("build", "run"))
     depends_on("py-pyproj@2.2.0:", type=("build", "run"), when="@0.7:")
     depends_on("py-pyproj@2.6.1.post1:", type=("build", "run"), when="@0.11:")
     depends_on("py-pyproj@3.3.0:", type=("build", "run"), when="@0.14:")
     depends_on("py-packaging", type=("build", "run"), when="@0.11:")
+    # Only for versions 0.14.4+ ? Or was it forgotten in pyproject.toml for 0.14.0 to 0.14.3?
+    depends_on("py-numpy@1.22:", type=("build", "run"), when="@0.14.4:")
+    # Only for versions 0.x.y
+    depends_on("py-fiona", type=("build", "run"), when="@:0.99")
+    depends_on("py-fiona@1.8:", type=("build", "run"), when="@0.9:0.99")
+    depends_on("py-fiona@1.8.21:", type=("build", "run"), when="@0.14:0.99")
+    # Only for versions 1.x.y
```

Comment:
It makes it much easier to review if deps are sorted:

1. in the same order as in pyproject.toml, and
2. from newest to oldest

Reply:
I moved the fiona section as it was from above to where it is now, because it is a historic dependency and replaced by py-gogrio for newer versions. I didn't change the order of the py-fiona dependencies at all. Should I move this back up where it was, together with the py-ogrio option? If I change the order for fiona I should change the order for all other existing dependencies as well.

Reply:
I changed the order of the dependencies to match the pyproject.toml order better. I didn't sort the versions from newest to oldest, since the existing geopandas file doesn't do that (but I can add that if you insist on it). I left py-fiona in the same place where py-ogrio is now, because that's where it is in earlier versions of pyproject.toml.